src/Profilers/MonitorProfiler/Communication/CommandServer.cpp

```diff
@@ -75,25 +79,39 @@ void CommandServer::ListeningThread()
         if (FAILED(hr))
         {
             _logger->Log(LogLevel::Error, _LS("Unexpected error when receiving data: 0x%08x"), hr);
+            // Best-effort shutdown, ignore the result.
+            client->Shutdown();
             continue;
         }
 
+        bool doEnqueueMessage = true;
+        hr = _validateMessageCallback(message);
+        if (FAILED(hr))
+        {
+            _logger->Log(LogLevel::Error, _LS("Failed to validate message: 0x%08x"), hr);
+            doEnqueueMessage = false;
+        }
+
         *reinterpret_cast<HRESULT*>(response.Payload.data()) = hr;
```

Comment:
Do we show the HR response on the client logs for this? If not we probably should to make it easier to figure out failures.